src/main/java/com/epam/grid/engine/provider/utils/slurm/healthcheck/ShowConfigCommandParser.java

```diff
@@ -0,0 +1,154 @@
+            throw new GridEngineException(HttpStatus.INTERNAL_SERVER_ERROR,
+                    String.format("Unexcpected error occurred during health check. "
+                                    + "%nexitCode = %d %nstdOut: %s %nstdErr: %s",
+                            commandResult.getExitCode(), commandResult.getStdOut(), commandResult.getStdErr())
+            );
+        }
+    }
+
+    private static boolean checkUnexpectedError(final CommandResult commandResult) {
+        return commandResult.getExitCode() != 0 && !commandResult.getStdErr().isEmpty();
+    }
+
+    private static StatusInfo parseStatusInfo(final List<String> stdOut) {
+        final long statusCode = parseStatusCode(stdOut);
+        return StatusInfo.builder()
+                .code(statusCode)
+                .status(getStatus(statusCode))
+                .info(parseInfo(stdOut))
+                .build();
+    }
+
+    private static long parseStatusCode(final List<String> stdOut) {
+        final long statusCode;
+        final List<String> statusString = List.of(stdOut.get(stdOut.size() - 1)
+                .replaceAll(SPACES_REGEX, SPACE)
+                .split(SPACE));
+        final String status = statusString.get(statusString.size() - 1);
+        switch (status) {
+            case UP_STATE:
+                statusCode = 0L;
+                break;
+            case DOWN_STATE:
+                statusCode = 2L;
+                break;
+            default:
+                statusCode = NOT_PROVIDED;
+                break;
+        }
+        return statusCode;
+    }
+
+    private static GridEngineStatus getStatus(final long status) {
+        return GridEngineStatus.getById(status)
+                .orElseThrow(() -> new GridEngineException(HttpStatus.INTERNAL_SERVER_ERROR,
+                        "Not valid status value provided: " + status));
+    }
+
+    private static String parseInfo(final List<String> stdOut) {
+        return stdOut.get(stdOut.size() - 1).trim();
+    }
+
+    private static LocalDateTime getCheckTime(final List<String> stdOut) {
+        return stdOut.stream().filter(out -> out.contains(CONFIGURATION_STRING)).findAny()
+                .map(start -> start.replaceAll(SPACES_REGEX, SPACE))
+                .map(start -> start.replace(CONFIGURATION_STRING, EMPTY_STRING))
+                .map(String::trim)
+                .map(start -> tryParseStringToLocalDateTime(start, DateTimeFormatter.ofPattern(SLURM_DATE_FORMAT)))
+                .orElseThrow(() -> new GridEngineException(HttpStatus.INTERNAL_SERVER_ERROR,
+                        "check time wasn't found in stdOut, stdOut: " + stdOut));
+    }
+
+    private static LocalDateTime getStartTime(final List<String> stdOut) {
+        return stdOut.stream().filter(out -> out.contains(BOOT_TIME)).findAny()
+                .map(bootTimeStr -> {
+                    final String[] bootTimeKeyValue = bootTimeStr.split(EQUAL_SIGN);
```

Comment:
Try to avoid complex blocks, when using Stream API - split them into a call chain, where possible. It makes code more compact and easy to read:

```java
...
.map(bootTimeStr -> /* split by equal */)
.map(keyValuePair -> /* return key or empty string */)
...
```

Reply:
done